library/src/main/java/com/novoda/downloadmanager/lib/DownloadReadyChecker.java

```diff
@@ -53,7 +55,7 @@ private boolean isDownloadManagerReadyToDownload(FileDownloadInfo downloadInfo)
             case DownloadStatus.WAITING_TO_RETRY:
                 // download was waiting for a delayed restart
                 final long now = systemFacade.currentTimeMillis();
-                return downloadInfo.restartTime(now) <= now;
+                return restartTimeCreator.restartTime(downloadInfo, now) <= now;
```

Comment:
whatever, all I'm saying is the behaviour can now be moved into that class `return restartTimer.shouldRestart(downloadInfo);`